picard/ui/scripteditor.py

```diff
@@ -0,0 +1,377 @@
+# -*- coding: utf-8 -*-
+#
+# Picard, the next-generation MusicBrainz tagger
+#
+# Copyright (C) 2021 Bob Swift
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+
+from functools import partial
+import os.path
+
+from PyQt5 import (
+    QtCore,
+    QtWidgets,
+)
+
+from picard.config import (
+    TextOption,
+    get_config,
+)
+from picard.const import DEFAULT_FILE_NAMING_FORMAT
+from picard.file import File
+from picard.script import (
+    ScriptError,
+    ScriptParser,
+)
+from picard.util.settingsoverride import SettingsOverride
+
+from picard.ui import (
+    PicardDialog,
+    SingletonDialog,
+)
+from picard.ui.options import OptionsPage
+from picard.ui.options.scripting import (
+    ScriptCheckError,
+    ScriptingDocumentationDialog,
+)
+from picard.ui.ui_scripteditor import Ui_ScriptEditor
+
+
+PRESET_SCRIPTS = [
```

Comment:
This list will be editable (ideally).
So I think we should get rid of the dummy entry (it should be the current user script, named `your script` or whatever).
Presets should be appended to it (not modifiable).
The combo box could be made editable to let user change titles.

We should perhaps objectify each item:

```python
class FileNamingScript:
   def __init__(self, script='', title=None, readonly=False):
       if title is None:
           title = N_('default filenaming script title')
       self.title = title
       self.script = script
       self.readonly = readonly   # for presets
```

Just a bunch of ideas